core/dbt/contracts/graph/nodes.py

```diff
@@ -1400,6 +1401,12 @@ def same_contents(self, old: Optional["Metric"]) -> bool:
             and True
         )
 
+    def add_input_measure(self, input_measure: MetricInputMeasure) -> None:
```

Comment:
Do we need a mutating method? An alternative would be to just use a `set` in `_process_metric_node` for constructing the list, and then at the end of the function convert via `type_params.input_measures=list(measures_set)`. I guess the drawback would be if anywhere else in the code base wanted to add additional measures to the list after the fact, but I'm not sure that's even a workflow we'd want to encourage or support.